labs/2-perpetual-exchange/2-trading.md

```diff
@@ -0,0 +1,97 @@
+
+When entering a position, whether Long or Short, it hinges on the available liquidity within the JLP pool.
+
+- Long Positions: For SOL, ETH, WBTC, it relies on the availability of the respective assets within the JLP pool.
+- Short Positions: In the case of SOL, ETH, WBTC, it depends on the availability of stablecoin assets, specifically USDC and USDT.
+
+## Managing Positions
+
+Upon opening a position, it becomes accessible in your `Positions` list. To adjust leverage and liquidation price, click on `Edit`. This feature enables efficient management of your positions.
+
+When initiating a position or depositing collateral, a snapshot of your collateral's USD value is captured. For instance, if your collateral is 1 SOL and SOL's price is 40 USD, your collateral is valued at 40 USD, remaining constant despite SOL price fluctuations.
+
+Profit and loss are proportionate to your position size. For instance, a long SOL position of 1,000 USD with a 10% SOL price increase yields a profit of 100 USD, while a 10% decrease results in a 100 USD loss. Conversely, a short position with a 10% SOL price decrease garners a 100 USD profit, but a 10% increase incurs a 100 USD loss.
+
+Leverage for a position is represented as ```Size (position size) / Collateral (position collateral)```.
+
+## Closing a Position
+
+You can close a position by clicking on the `Close` button in the position row.
+
+- **Long Positions:** Profits from long positions are disbursed in the asset you are longing. For instance, if you've gone long on SOL, your profits will be received in SOL.
+
+- **Short Positions:** Profits from short positions are paid out in the same stablecoin used to open the position, such as USDC or USDT.
+
+*Note: You also have the option to select a specific token for closure and receive the assets accordingly, taking market conditions into account. Jupiter will assist in seamlessly swapping into your desired token. This feature provides users with additional control and optimization over their trading outcomes.*
+
+
+## Take-Profit / Stop-Loss Orders
+
+You can set take-profit and stop-loss trigger orders by selecting the `Add TP` or `Add SL` button in the position row.
+
+Once you create a trigger order, it will be displayed in your position's row under the `Take Profit` and `Stop Loss` columns. You have the flexibility to edit the order and adjust the trigger price as needed.
+
+In the event of manually closing a position, the associated trigger orders will be automatically canceled and closed, providing a seamless experience when opening new future positions.
+
+*Note: While orders are set, execution is not guaranteed. Several factors, including the mark price (an aggregate of exchange prices), reaching the specified price, and the order's size exceeding the remaining position size, can influence the execution outcome.*
```

Comment:
TPSL currently will only close the whole position, order's size won't exceed the remaining position size